voter-dapp/src/RetrieveRewards.js

```diff
@@ -14,13 +15,13 @@ function getOrCreateObj(containingObj, field) {
   return containingObj[field];
 }
 
-function useRetrieveRewardsTxn(retrievedRewardsEvents, revealedVoteEvents, priceResolvedEvents, votingAccount) {
+function useRetrieveRewardsTxn(retrievedRewardsEvents, revealedVoteEvents, priceRequests, votingAccount) {
```

Comment:
Maybe this is just me, but I was a bit confused by the name `priceRequests`. I didn't realize that `priceRequests` and `revealedVoteEvents` were 1-to-1 mapped. I think it'd be a bit cleaner if you collapsed all of the data into one object OR you indicated that `priceRequests` was some sort of `revealedVote`-mapping-to-`priceRequests`. 

Additionally, is `retrievedRewardsEvents` used anywhere in this method?

Reply:
>Additionally, is retrievedRewardsEvents used anywhere in this method?

Ah I see it is used. nevermind

Reply:
Yeah, a lot of this is just because I'm trying to retrofit this method to a new way of computing these values.

Collapsing retrievedRewardsEvents and revealedVoteEvents is what a lot of this method is doing. There are a lot of subtleties in doing that -- one being that revealedVoteEvents can have multiple votes on the same price request in the case that the earlier votes don't hit the GAT. Some revealed votes don't exist in retrieved rewards (for obvious reasons). Merging the two requires you to create a completely distinct data structure that can handle all the subtle partial overlaps. That's a lot of what this method is trying to handle (there used to be a third set of overlapping events, too, which made it more necessary.

I agree that having priceRequests be 1:1 mapped with revealed votes is a little confusing. I could technically make them part of the same array pretty easily. Wasn't sure if it would be more confusing to dump price request data into the revealedVoteEvent array. Do you think mutating those event structs would make things simpler? I'm happy to do that -- it wouldn't change the structure too much.

Thoughts?

Reply:
I do think putting them into the same structure is more sustainable. This way no one can unintentionally cause the two arrays to be misaligned

Reply:
@nicholaspai done. PTAL.